python/src/bindings.cpp

```diff
@@ -464,14 +463,15 @@ as passing a batch of vectors helps avoid Python's Global Interpreter Lock.
   ////////////////////////////////////////////////////////////////////////////////////////////////////
   index.def(
       "query",
-      [](Index &index, py::array_t<float> input, size_t k = 1,
+      [](Index &index, nb::ndarray<float> &input, size_t k = 1,
```

Comment:
These changes from `py::array_t` to `nb::ndarray` are a breaking change, the Python code that uses List objects won't work anymore, e.g.
```
neighbors, distances = index.query([1, 2, 3, 4, 5], k=2)
```

Does nanobind have an array type we can use?

Reply:
```python
>>> # Create an empty Index object that can store vectors:
>>> index = Index(Space.Euclidean, num_dimensions=5)
>>> id_a = index.add_item([1, 2, 3, 4, 5])
>>> id_b = index.add_item([6, 7, 8, 9, 10])
>>> 
>>> print(id_a)  # => 0
0
>>> print(id_b)  # => 1
1
>>> 
>>> # Find the two closest elements:
>>> neighbors, distances = index.query([1, 2, 3, 4, 5], k=2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: query(): incompatible function arguments. The following argument types are supported:
    1. query(self, vectors: ndarray[dtype=float32], k: int = 1, num_threads: int = -1, query_ef: int = -1) -> typing.Tuple[numpy.ndarray[dtype=uint64], numpy.ndarray[dtype=float32]]

Invoked with types: voyager.voyager_ext.FloatIndex, list, kwargs = { k: int }

```

Reply:
But none of the Python tests fail because we are creating numpy arrays in them instead of Lists